include/nbl/video/IDriverMemoryAllocation.h

```diff
@@ -141,8 +143,12 @@ class IDriverMemoryAllocation : public virtual core::IReferenceCounted
             currentMappingAccess = access;
         }
 
-        IDriverMemoryAllocation() : mappedPtr(nullptr), mappedRange(0,0), currentMappingAccess(EMCAF_NO_MAPPING_ACCESS) {}
+        IDriverMemoryAllocation(const ILogicalDevice* originDevice)
+            : m_originDevice(originDevice), mappedPtr(nullptr), mappedRange(0,0),
+            currentMappingAccess(EMCAF_NO_MAPPING_ACCESS)
+        {}
 
+        const ILogicalDevice* m_originDevice = nullptr;
```

Comment:
@achalpandeyy remind me, did we decide AGAINST `IDriverMemoryAllocation` keeping a smartrefctd pointer back to the `ILogicalDevice`?

AFAIK we only decided to not keep smartpointers to `IDriverMemoryAllocation`